openmc/mgxs/mgxs.py

```diff
@@ -0,0 +1,5909 @@
+        Filters will be provided as a ragged 2D list of openmc.Filter objects.
+
+        Parameters
+        ----------
+        filters : Iterable of Iterable of openmc.Filter
+            Ragged 2D list of openmc.Filter objects for the energy and spatial
+            domains. The angle filters will be added to the list.
+
+        Returns
+        -------
+        Iterable of Iterable of openmc.Filter
+            Ragged 2D list of openmc.Filter objects for the energy and spatial
+            domains with the angle filters added to the list.
+
+        """
+
+        if self.num_polar > 1 or self.num_azimuthal > 1:
+            # Then the user has requested angular data, so create the bins
+            pol_bins = np.linspace(0., np.pi, num=self.num_polar + 1,
+                                   endpoint=True)
+            azi_bins = np.linspace(-np.pi, np.pi, num=self.num_azimuthal + 1,
+                                   endpoint=True)
+
+            for filt in filters:
+                filt.insert(0, openmc.PolarFilter(pol_bins))
+                filt.insert(1, openmc.AzimuthalFilter(azi_bins))
+
+        return filters
+
+    def _squeeze_xs(self, xs):
+        """Remove dimensions which are not needed from a cross section array
+        due to user options. This is used by the openmc.Mgxs.get_xs(...) method
+
+        Parameters
+        ----------
+        xs : np.ndarray
+            Cross sections array with dimensions to be squeezed
+
+        Returns
+        -------
+        np.ndarray
+            Squeezed array of cross sections
+
+        """
+
+        # numpy.squeeze will return a ValueError if the axis has a size
+        # greater than 1, to avoid this we will try each axis one at a
+        # time to preclude the ValueError.
+        initial_shape = len(xs.shape)
+        for axis in range(initial_shape - 1, -1, -1):
+            if axis not in self._dont_squeeze and xs.shape[axis] == 1:
+                xs = np.squeeze(xs, axis=axis)
+        return xs
+
+    def _df_convert_columns_to_bins(self, df):
+        """This method converts all relevant and present DataFrame columns from
+        their bin boundaries to the index for each bin. This method operates on
+        the DataFrame, df, in place. The method returns a list of the columns
+        in which it has operated on.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            A Pandas DataFrame containing the cross section data.
+
+        Returns
+        -------
+        columns : Iterable of str
+            Names of the re-named and re-valued columns
+
+        """
+        # Override polar and azimuthal bounds with indices
+        if self.num_polar > 1 or self.num_azimuthal > 1:
+            # First for polar
+            bins = np.linspace(0., np.pi, self.num_polar + 1, True)
+            _df_column_convert_to_bin(df, 'polar low', 'polar bin', bins)
+            del df['polar high']
```

Comment:
Might be better to use `.pop()` here instead of `del` just in case `polar high` doesn't exist.